DEVELOPMENT.md

```diff
@@ -115,3 +139,49 @@ bazel build third_party/chromium
 
 # Lastly, put the address returnd by the web server into the Chromium.
 ```
+
+#### Running webapp tests
+
+tl;dr. Use `ibazel test` for regular work (supports limiting the test amount with
+`fit/fdescript` and `console.log` usage) and `ibazel run` for karma console breakpoint
+debugging.
+
+We use karma testing chromium target for our UI unit tests in webapp. However,
+depending on the environments (OSS or internal TB.corp), the commands we use and
+the functionalities are slightly different.
+
+1.  Just run all webapp tests. The job stops after finished. `console.log` is not
+    supported. Not handy on development.
+
+    ```shell
+    (tf)$ bazel test //tensorboard/webapp:karma_test_chromium-local
+    ```
+
+2.  Using `ibazel` to auto detect the file changes and use target
+    `karma_test_chromium-local` for running on *all* tests. Practically
+    `console.log` and `fit/fdescript` (which areused to narrow down the test amount) are
+    working.
+
+    ```shell
+    (tf)$ ibazel test //tensorboard/webapp:karma_test_chromium-local --test_output=all
+    ```
+
+    *   `--test_output=all`: for displaying number of tests if using '`fit`'.
+
+3.  To run on a specific test, we can change the target (with `chromium-local`
+    surfix). For example, run notification center tests on `notification_center_test` target:
+
+    ```shell
+    (tf)$ ibazel test //tensorboard/webapp/notification_center:notification_center_test_chromium-local
+    ```
+
+4.  For having a karma console to set break points for debugging purpose, use
+    `ibazel run`. Access the karma console at port `9876` and click 'DEBUG'
+    button. For example, `http://<YOUR_SERVER_ADDRESS>:9876/`.
```

Comment:
I would also mention that you have to look at the browser developer console to see the results. It's not rendered in the browser window. This always confuses me.

Reply:
Ah, added.